modules/basic-skills/src/backend/choice_ndu.ts

```diff
@@ -0,0 +1,148 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import yn from 'yn'
+
+const generateFlowNdu = async (data: any): Promise<sdk.FlowGenerationResult> => {
+  const { randomId } = data
+
+  const hardRetryLimit = 10
+  const nbMaxRetries = Math.min(Number(data.config.nbMaxRetries), hardRetryLimit)
+  const repeatQuestion = yn(data.config.repeatChoicesOnInvalid)
+
+  const sorrySteps = []
+
+  if (data.invalidContentId && data.invalidContentId.length >= 3) {
+    sorrySteps.push({
+      type: sdk.NodeActionType.RenderElement,
+      name: `#!${data.invalidContentId}`
+    })
+  }
+
+  if (repeatQuestion) {
+    sorrySteps.push({
+      type: sdk.NodeActionType.RenderElement,
+      name: `#!${data.contentId}`,
+      args: { skill: 'choice' }
+    })
+  }
+
+  const nodes: sdk.SkillFlowNode[] = [
+    {
+      name: 'entry',
+      onEnter: [
+        {
+          type: sdk.NodeActionType.RenderElement,
+          name: `#!${data.contentId}`,
+          args: { skill: 'choice' }
+        }
+      ],
+      onReceive: [],
+      next: [{ condition: 'true', node: 'wait' }]
+    },
+    {
+      name: 'wait',
+      onReceive: [],
+      next: [{ condition: 'true', node: 'wait' }]
```

Comment:
Yup it is, because I want to keep him in that flow. The on receive will prevent the infinite loop, and any triggers that activates will send the user to a different node

Reply:
Oh, right. Can you add a comment if not done already